build.proj

```diff
@@ -52,7 +52,7 @@
   <Import Project="build/Compile.targets" />
   <Import Project="build/Package.targets" />
   <Import Project="build/Test.targets" />
-  <Import Project="build/Publish.targets" />
+  <Import Condition=" '$(GeneratePropsFile)' != 'true' " Project="build/Publish.targets" />
```

Comment:
Instead of this I would suggest doing the condition in publish.targets on the existence of the targets file you are trying to import there.

Reply:
Good idea; done...